components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -0,0 +1,219 @@
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.constants.SQLConstants;
+import org.wso2.carbon.apimgt.impl.utils.GatewayArtifactsMgtDBUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GatewayArtifactsMgtDAO {
+    private static final Log log = LogFactory.getLog(GatewayArtifactsMgtDAO.class);
+    private static GatewayArtifactsMgtDAO INSTANCE = null;
+
+    /**
+     * Method to get the instance of the GatewayArtifactsMgtDAO.
+     *
+     * @return {@link GatewayArtifactsMgtDAO} instance
+     */
+    public static GatewayArtifactsMgtDAO getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new GatewayArtifactsMgtDAO();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Add details of the APIs published in the Gateway
+     *
+     * @param APIId        - UUID of the API
+     * @param APIName      - Name of the API
+     * @param version      - Version of the API
+     * @param tenantDomain - Tenant domain of the API
+     * @throws APIManagementException if an error occurs
+     */
+    public void addGatewayPublishedAPIDetails(String APIId, String APIName, String version, String tenantDomain)
+            throws APIManagementException {
+
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
+                PreparedStatement statement = connection.prepareStatement(SQLConstants.ADD_GW_PUBLISHED_API_DETAILS)) {
+            statement.setString(1, APIId);
+            statement.setString(2, APIName);
+            statement.setString(3, version);
+            statement.setString(4, tenantDomain);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            handleException("Failed to add API details for " + APIName, e);
+        }
+    }
+
+    /**
+     * Add or update details of the APIs published in the Gateway
+     *
+     * @param APIId        - UUID of the API
+     * @param gatewayLabel - Published gateway's label
+     * @param bais         - Byte array Input stream of the serializide gatewayAPIDTO
+     * @param streamLength - Length of the stream
+     * @throws APIManagementException if an error occurs
+     */
+    public void addGatewayPublishedAPIArtifacts(String APIId, String gatewayLabel, ByteArrayInputStream bais,
+            int streamLength, String gatewayInstruction, String query)
+            throws APIManagementException {
+
+        try (Connection connection = GatewayArtifactsMgtDBUtil.getArtifactSynchronizerConnection();
```

Comment:
fix all the places as above except GET

Reply:
https://github.com/wso2/carbon-apimgt/pull/8828/commits/d501401001c713712bb50c42b40aa2afa9326545